File: chia-rest/app/entities/blockchain_state.py
""""Block chain state definitions."""


from dataclasses import dataclass

from app.entities.base import BaseEntity


@dataclass(kw_only=True)
class BlockChainState(BaseEntity):
    """Data class representation of the blockchain state."""

    node_id: str
    height: int
    block_max_cost: int
    difficulty: int
    space: int
    syncing: bool
    sync_progress_height: int
    sync_tip_height: int

File: ChiaMonitor/app/src/main/java/com/example/chiamonitor/domain/samples/BlockchainState.kt
package com.example.chiamonitor.domain.samples

import com.example.chiamonitor.domain.BlockchainState


val sampleSyncingBlockchainState =  BlockchainState(
    nodeId = "b3d9de85d29931c10050b56c7afb91c99141943fc81ff2d1a8425e52be0d08ab",
    height = 3_326_213,
    blockMaxCost = 11000000000,
    difficulty = 1976,
    space = 251933217250036449,
    syncing = true,
    syncProgressHeight = 2_123_456,
    syncTipHeight = 3_326_213,
)


val sampleSyncedBlockchainState = BlockchainState(
    nodeId = "b3d9de85d29931c10050b56c7afb91c99141943fc81ff2d1a8425e52be0d08ab",
    height = 4_326_213,
    blockMaxCost = 11000000000,
    difficulty = 1976,
    space = 251933217250036449,
    syncing = false,
    syncProgressHeight = 0,
    syncTipHeight = 0,
)
